packages/kbn-alerts-ui-shared/src/rule_form/rule_definition/rule_alert_delay.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback } from 'react';
+import { EuiFieldText, EuiFormRow, EuiIconTip } from '@elastic/eui';
+import type { SanitizedRule, RuleTypeParams } from '@kbn/alerting-types';
+import {
+  ALERT_DELAY_TITLE_PREFIX,
+  ALERT_DELAY_TITLE_SUFFIX,
+  ALERT_DELAY_HELP_TEXT,
+  ALERT_DELAY_TITLE,
+} from '../translations';
+import { RuleFormErrors } from '../types';
+
+const INTEGER_REGEX = /^[1-9][0-9]*$/;
+
+export interface RuleAlertDelayProps {
+  alertDelay?: SanitizedRule<RuleTypeParams>['alertDelay'];
+  errors?: RuleFormErrors;
+  onChange: (property: string, value: unknown) => void;
+}
+
+export const RuleAlertDelay = (props: RuleAlertDelayProps) => {
+  const { alertDelay, errors = {}, onChange } = props;
+
+  const onAlertDelayChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      if (!e.target.validity.valid) {
+        return;
+      }
+      const value = e.target.value;
+      if (value === '') {
+        onChange('alertDelay', null);
+      } else if (INTEGER_REGEX.test(value)) {
+        const parsedValue = parseInt(value, 10);
+        onChange('alertDelay', { active: parsedValue });
+      }
+    },
+    [onChange]
+  );
+
+  return (
+    <EuiFormRow
+      fullWidth
+      label={ALERT_DELAY_TITLE}
+      isInvalid={errors.alertDelay?.length > 0}
```

Comment:
I can give it a type, but they have to be collected in the `errors` object. But the key should just map to the corresponding rule property: I.E `rule.alertDelay` -> `errors.alertDelay`, etc.

Reply:
I was thinking something like `<RuleAlertDelay errors={errors.alertDelay}>` and the type of `errors` can be whatever the type of `errors.alertDelay`.

Reply:
In a future PR all of the errors will come from the context, so `<RuleAlertDelay errors={errors.alertDelay}>` wont really apply, since it won't come from the props anymore.